includes/utilities.inc

```diff
@@ -47,58 +47,396 @@ function islandora_openseadragon_construct_clip_url($params, $download = FALSE)
+ *   source to be rendered.
+ *   - identifier: A URL where the datastream image can be accessed.
+ *   - width: The width of the image in pixels if known.
+ *   - height: The height of the image in pixels if known.
+ *   - tileWidth: The width of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileHeight: The height of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileOverlap: The number of pixels each tile is expected to overlap
+ *     touching tiles.
+ *   - minLevel: The minimum level to attempt to load.
+ *   - maxLevel: The maximum level to attempt to load.
+ */
+function islandora_openseadragon_iiif_tile_source(AbstractDatastream $datastream) {
+  $base_url = variable_get('islandora_openseadragon_iiif_url', 'loris');
+  $pid = $datastream->parent->id;
+  $dsid = $datastream->id;
+  // Currently Loris works straight out of fedora only.
+  return url("{$base_url}/{$pid}/datastreams/{$dsid}/", array(
+    'absolute' => TRUE,
+  ));
+}
+
+/**
+ * Image metadata pertinent to the Djatoka TileSources implementation.
+ *
+ * @param AbstractDatastream $datastream
+ *   Gets the metadata for viewing the given datastream, it assumes that the
+ *   datastream is an image and it's format is supported by Adore Djatoka.
+ *
+ * @return array
+ *   An associative array with the required metadata for AdoraDjatoka tile
+ *   source to be rendered.
+ *   - identifier: A URL where the datastream image can be accessed.
+ *   - width: The width of the image in pixels if known.
+ *   - height: The height of the image in pixels if known.
+ *   - tileWidth: The width of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileHeight: The height of the tiles to assumed to make up each pyramid
+ *     layer in pixels.
+ *   - tileOverlap: The number of pixels each tile is expected to overlap
+ *     touching tiles.
+ *   - minLevel: The minimum level to attempt to load.
+ *   - maxLevel: The maximum level to attempt to load.
+ */
+function islandora_openseadragon_djatoka_tile_source(AbstractDatastream $datastream) {
+  module_load_include('inc', 'islandora', 'includes/authtokens');
+  $pid = $datastream->parent->id;
+  $dsid = $datastream->id;
+  $token = islandora_get_object_token($pid, $dsid, 2);
   return array(
-    // XXX: Using drupal_http_build_query to avoid changing the underlying
-    // functionality in the solution packs that currently implement.
-    'original_params' => drupal_http_build_query($allowed_djatoka_params),
-    'djatoka_image_url' => $url,
-    'dimensions' => $dimensions,
+    'identifier' => url("islandora/object/{$pid}/datastream/{$dsid}/view", array(
+                    'absolute' => TRUE,
+                    'query' => array(
+                      'token' => $token,
+                    ),
+                  )),
+    'width' => NULL,
+    'height' => NULL,
+    'tileWidth' => 256,
+    'tileHeight' => 256,
+    'tileOverlap' => 2,
```

Comment:
Ultimately I don't have a good reason for removing it. That being said anytime I played with those values or the overlap it seemed to break the rendering of the images. Also, somehow I feel it should be set on a tile source by tile source basis, rather than for every image in the repository, but that's just a gut feeling. I'll add them back in.